projects/optic/src/commands/capture/capture.ts

```diff
@@ -197,43 +202,71 @@ const getCaptureAction =
     }
 
     // make requests
-    const concurrency = captureConfig.config?.request_concurrency || 5;
-    const requests = makeRequests(
-      captureConfig.requests,
-      proxyUrl,
-      concurrency
-    );
-
     const captures = new GroupedCaptures(
       trafficDirectory,
       getEndpointsFromSpec(spec.jsonLike)
     );
-    const harEntries = HarEntries.fromProxyInteractions(proxyInteractions);
 
-    await Promise.all(requests)
-      .catch((error) => {
-        console.error(error);
-      })
-      .finally(() => {
-        // stop the proxy
-        sourcesController.abort();
-        // stop the app server
-        if (!options.serverOverride && captureConfig.server.command) {
-          process.kill(-app.pid!);
-        }
-      });
+    let requestsPromise: Promise<any> = Promise.resolve();
+    let reqCmdPromise: Promise<void> = Promise.resolve();
+    try {
+      if (captureConfig.requests) {
+        const requests = makeRequests(
+          captureConfig.requests,
+          proxyUrl,
+          captureConfig.config?.request_concurrency || 5
+        );
+        requestsPromise = Promise.all(requests);
+      }
+
+      if (captureConfig.requests_command) {
+        const cmd = commandSplitter(captureConfig.requests_command.command);
+        const proxyVar =
+          captureConfig.requests_command.proxy_variable || 'OPTIC_PROXY';
+        process.env[proxyVar] = proxyUrl;
+        const reqCmd = spawn(cmd.cmd, cmd.args, {
+          detached: true,
+          shell: true,
+        });
+
+        reqCmdPromise = new Promise((resolve, reject) => {
+          reqCmd.on('exit', (code) => {
+            if (code === 0) {
+              resolve();
+            } else {
+              reject();
+            }
+          });
+        });
+        reqCmd.stderr.on('data', (data) => {
+          logger.error(data.toString());
+        });
+      }
+    } catch (error) {
+      logger.error(error);
+    } finally {
+      await Promise.all([requestsPromise, reqCmdPromise]);
```

Comment:
I think this should go right above the catch block (i.e. we should wait for it as part of the `try`. `await`ing a Promise will throw an error, which means we want any potential error to be thrown in the `try` part